### PR TITLE
[CDAP-17063] Add additional functionalities to improve user experience (RequestHistoryTab)

### PR DESCRIPTION
**Pull request by jennac3**
Screen Shot 2020-07-09 at 8 23 36 AM

- Hovering over a request log, we can see additional icons, which allow us to
    1. view timestamp of the log
    2. delete request log
- Search over request logs
-  Enable / disable save mode. When `saveRequests` mode is on, we save requests.
- Clear all request logs

### CODE REVIEW

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestActionDialogs/ClearDialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    notifyRequestClear();
+    handleClose();
+  };
+
+  return (
+    <ConfirmationModal
+      isOpen={open}
+      headerTitle={'Clear all your request history'}
+      confirmationElem={
+        'Are you sure you want to clear all of your request history from your CDAP Account?'
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestActionDialogs/ClearDialog/index.tsx]: I wonder if we should leave off the reference to the CDAP account, so this message makes sense in CDF as well, i.e. "Are you sure you want to clear all your request history?"

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestActionDialogs/DeleteDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    notifyRequestDelete(id);
+    handleClose();
+  };
+
+  return (
+    <ConfirmationModal
+      isOpen={open}
+      headerTitle={'Delete your request history'}
+      confirmationElem={
+        'Are you sure you want to delete your request history from your CDAP Account?'
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestActionDialogs/DeleteDialog/index.tsx]: Same comment as above!

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            title={renderTimestamp(request.requestID)}
+            placement="right"
+            className={classes.requestActionButton}
+          >
+            <ScheduleIcon />
+          </Tooltip>
+        </div>
+        <div
+          className={classnames(classes.requestMethod, {
+            [classes.getMethod]: request.method === RequestMethod.GET,
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx:181]: use enum?

Comment (jennac3) [reply cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx:181]: Which variable are you referring to?

Comment (yukiej) [reply cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx:181]: I meant the request methods, but it looks like you are already doing it so maybe I was looking in the wrong place or saw something outdated. So no change needed!

**Comment by jennac3** [comment]
Screen Shot 2020-07-21 at 3 37 56 PM


@leacd this is the new UI!

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+      '&:hover': {
+        backgroundColor: theme.palette.grey[700],
+        '& $requestActionButton': {
+          visibility: 'visible',
+        },
+      },
+    },
+    selectedRequestRow: {
+      backgroundColor: `${theme.palette.blue[500]} !important`,
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx:49]: is the !important necessary?

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      wordWrap: 'break-word',
+      textAlign: 'left',
+      alignSelf: 'center',
+      textTransform: 'lowercase',
+      fontSize: '10px',
+      display: 'inline-block',
+      lineHeight: '1.3',
+    },
+    selectedRequestPath: {
+      color: `${theme.palette.orange[50]} !important`,
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx]: same here regarding the !important

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestSearch/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import SearchIcon from '@material-ui/icons/Search';
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      padding: '2px 4px',
+      display: 'flex',
+      alignItems: 'center',
+      width: '100%',
+      borderTop: `0.5px solid ${theme.palette.grey[300]}`,
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestSearch/index.tsx]: should avoid float values for pixel size

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx [hunk 7]
```diff
@@ -123,46 +123,37 @@ const StyledExpansionPanel = withStyles((theme) => ({
...
 }))(ExpansionPanel);
 
+const StyledExpansionPanelSummary = withStyles({
+  expandIcon: {
+    order: -1,
+    paddingRight: '10px',
+  },
+  root: {
+    height: '35px !important',
+    padding: '0px !important',
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx]: withStyles should already overwrite these values  for ExpansionPanelSummary. Are the !important necessary?

Comment (jennac3) [reply cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx]: done

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestSearch/index.tsx (hunk 6 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestSearch/index.tsx]: done

File: cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx (hunk 4 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/RequestRow/index.tsx:49]: done

**Review by yukiej — APPROVED** [review]
lgtm!

**Pull request merged**